tests/snmp/test_snmp_phy_entity.py

```diff
@@ -0,0 +1,583 @@
+        assert transceiver_snmp_fact['entPhysHwVer'] == transceiver_info['hardware_rev']
+        assert transceiver_snmp_fact['entPhysFwVer'] == ''
+        assert transceiver_snmp_fact['entPhysSwVer'] == ''
+        assert transceiver_snmp_fact['entPhysSerialNum'] == transceiver_info['serial']
+        assert transceiver_snmp_fact['entPhysMfgName'] == transceiver_info['manufacturer']
+        assert transceiver_snmp_fact['entPhysModelName'] == transceiver_info['model']
+        assert transceiver_snmp_fact['entPhysIsFRU'] == REPLACEABLE if transceiver_info[
+                                                                           'is_replaceable'] == 'True' else NOT_REPLACEABLE
+        _check_transceiver_dom_sensor_info(transceiver_snmp_fact['oid'], snmp_physical_entity_info)
+
+
+def _check_transceiver_dom_sensor_info(transceiver_oid, snmp_physical_entity_info):
+    """
+    Check transceiver DOM sensor information in physical entity mib
+    :param transceiver_oid: Transceiver oid
+    :param snmp_physical_entity_info: Physical entity information from snmp fact
+    :return:
+    """
+    for index, sensor_oid_offset in enumerate(XCVR_SENSOR_OID_LIST):
+        expect_oid = transceiver_oid + sensor_oid_offset
+        assert expect_oid in snmp_physical_entity_info, 'Cannot find port sensor in physical entity mib'
+        sensor_snmp_fact = snmp_physical_entity_info[expect_oid]
+        assert sensor_snmp_fact['entPhysDescr'] is not None
+        assert sensor_snmp_fact['entPhysContainedIn'] == transceiver_oid
+        assert sensor_snmp_fact['entPhysClass'] == PHYSICAL_CLASS_SENSOR
+        assert sensor_snmp_fact['entPhyParentRelPos'] == index + 1
+        assert sensor_snmp_fact['entPhysName'] is not None
+        assert sensor_snmp_fact['entPhysHwVer'] == ''
+        assert sensor_snmp_fact['entPhysFwVer'] == ''
+        assert sensor_snmp_fact['entPhysSwVer'] == ''
+        assert sensor_snmp_fact['entPhysSerialNum'] == ''
+        assert sensor_snmp_fact['entPhysMfgName'] == ''
+        assert sensor_snmp_fact['entPhysModelName'] == ''
+        assert sensor_snmp_fact['entPhysIsFRU'] == NOT_REPLACEABLE
+
+
+@pytest.mark.disable_loganalyzer
+def test_turn_off_psu_and_check_psu_info(duthost, localhost, creds, psu_controller):
+    """
+    Turn off one PSU and check all PSU sensor entity being removed because it can no longer get any value
+    :param duthost: DUT host object
+    :param localhost: localhost object
+    :param creds: Credential for snmp
+    :param psu_controller: PSU controller
+    :return:
+    """
+    if not psu_controller:
+        pytest.skip('psu_controller is None')
+    psu_status = psu_controller.get_psu_status()
+    if len(psu_status) < 2:
+        pytest.skip('At least 2 PSUs required for rest of the testing in this case')
+
+    # turn on all PSU
+    for item in psu_status:
+        if not item['psu_on']:
+            psu_controller.turn_on_psu(item["psu_id"])
+            time.sleep(5)
```

Comment:
>time.sleep(5) [](start = 12, length = 13)

What if no sleep?

Reply:
The PSU may take a few seconds to power on, so I add a sleep here, this code is similar with some test cases in platform tests.

Reply:
Better to sleep only once after all PSU turned on?

---
In reply to: [532159395](https://github.com/Azure/sonic-mgmt/pull/2379#discussion_r532159395) [](ancestors = 532159395)